# **Google Data Analytics Capstone Project**

### ***Business Case : Cyclistic, Chicago based company***
D Roopa

# **Introduction** 

This project is a part of Google Data Analytics Professional certicate Business case study. The document for the case study is available on  [Google Data Analytics Capstone: Complete a Case Study course.](http://www.coursera.org/learn/google-data-analytics-capstone)

To answer the key business questions at all six steps of the data analysis process: Ask, prepare, process, analyze, share and act certain roadmap is followed:
*       Code, if required.
*       Guiding questions and their answers
*       Key tasks and deliverables as a checklist.

The following is the scenario:

 **Scenario**
 
  I am playing the role of junior data analyst working in the marketing analyst team at Cyclists, a bike-share company in Chicago. 
The company has two types of customers:
* Casual riders: Customers who purchase single-ride or full-day passes.
* Annual members/Cyclistic members : Customers with annual memberships.

There are three types of bikes mentioned in the dataset.
1. Classic bikes
2. Docked bikes
3. Electric bikes

The director of marketing believes the company’s future success depends on maximizing the number of annual memberships. Therefore, the team wants to understand how casual riders and annual members use Cyclistic bikes differently. From these insights, the team will design a new marketing strategy to convert casual riders into annual members. But first, Cyclistic executives must approve your recommendations, so they must be backed up with compelling data insights and professional data visualizations.
A bike-share program that features more than 5,800 bicycles and 600 docking stations. The majority of riders opt for traditional bikes; about 8% of riders use assistive options. Cyclistic users are more likely to ride for leisure, but about 30% use them to commute to work each day.


Three questions to guide the future marketing program:
1. How do annual members and casual riders use Cyclistic bikes differently?
2. Why would casual riders buy Cyclistic annual memberships?
3. How can Cyclistic use digital media to influence casual riders to become members?

# ASK

First step of data analysis process
 
Business task identification and consideration of key stakeholders.

1. Business task : Understand Annual and casual rider usage patterns to recommend marketing strategies to convert casual riders into annual members and how digital platforms could influence them.

2. Stakeholders :
Lily Moreno: The director of marketing and my manager
Cyclistic marketing analytics team: A data analysts team responsible to guide Cyclistic’s marketing strategy by collecting, analyzing, and reporting data.
Cyclistic executive team: The detail-oriented executive team who would decide whether to approve the recommended marketing program.


**1.** ***Guiding* *questions***

* What is the problem you are trying to solve?

To build casual rider and annual members profile and recommend best marketing strategies to convert casual riders into Cyclistics annual member.

* How can your insights drive business decisions?

With insights, the marketing team can confidently build strong marketing strategies to convert casual riders into more profitable annual members.

**2.** ***Key tasks***

[X] Identify the business task

[X] Consider key stakeholders

**3.** ***Deliverable***

[X] A clear statement of the business task

# Prepare

Second step in data analysis process:Data collection, identify how data is organized and determine data creadibility.


Public data of Cyclistic’s historical trip data is available on [Motivate International Inc.](http://divvy-tripdata.s3.amazonaws.com/index.html) under the [license](http://ride.divvybikes.com/data-license-agreement).

For this project I have downloaded zip file from [Motivate International Inc.] and added it to Kaggle dataset as [cyclistic-data ](http://www.kaggle.com/datasets/roopamoorthy/cyclistic-data) , to overcome the warning of the "dataset already available" because there are n no. of analysis carried out using same dataset which is  available on Kaggle. Therefore,I opted to "create with duplicates"  to upload data on Kaggle from my local drive.

For analysis purpose I have used 12 months of data , April 2020 to March 2021.
To get a brief look I used MS Excel and noticed each month's data is recorded in separate CSV files with fields such as ride id, rideable type, start and end time, start and end station, latitude, and longitude of the start and end stations.


**1.** **Guiding questions**

* **Where is your data located?**

The data is located in a  dataset of Kaggle.

* **How is the data organized?**

Data is a CSV file for each month with bike ride information such as ride ID, rideable type, start and end time, start and end station, latitude, and longitude of start and end stations.


* **Are there issues with bias or credibility in this data? Does your data ROCCC?**

Bias: No there are no issues of bias because the population of the dataset is Company's clients as bike riders.
Data also has full credibility for the same reason. Hence, It is ROCCC because it's reliable, Original, comprehensive, current and cited.

* **How are you addressing licensing, privacy, security, and accessibility?**

The data has been made available by Motivate International Inc. under the license.
 Besides, there is no personally identifiable information about riders.

* **How did you verify the data’s integrity?**

The CSV files for each month have the same number of columns and correct data type.

* **How does it help you answer your question?**

The data has some information about riders to help in answering key business questions.

* **Are there any problems with the data?**

More information about the riders on an individual level could be useful.

**2.** **Key tasks**

[x] Download data and store it appropriately.

[x] Identify how it’s organized.

[x] Sort and filter the data.

[x] Determine the credibility of the data.

**3.** **Deliverable**

[x] A description of all data sources used.

# Process

Third step in data analysis process : Choose a tool for data cleaning, check errors in data, and document the cleaning process.

For this project, I will use R language in Kaggle to import the dataset and check how the data is organized and ensure all the columns has the right data type.

**1. Code**

In [63]:
# To load tidyverse,skimr,dplyr,janitor and lubridate libraries

library(tidyverse)
library(skimr)
library(dplyr)
library(janitor)
library(lubridate)

In [64]:
# import CSV files for each month

April_2020 <- read.csv("../input/cyclistic-data/202004-divvy-tripdata.csv")
May_2020 <- read.csv("../input/cyclistic-data/202005-divvy-tripdata.csv")
June_2020 <- read.csv("../input/cyclistic-data/202006-divvy-tripdata.csv")
July_2020 <- read.csv("../input/cyclistic-data/202007-divvy-tripdata.csv")
August_2020 <- read.csv("../input/cyclistic-data/202008-divvy-tripdata.csv")
September_2020<- read.csv("../input/cyclistic-data/202009-divvy-tripdata.csv")
October_2020 <- read.csv("../input/cyclistic-data/202010-divvy-tripdata.csv")
November_2020 <- read.csv("../input/cyclistic-data/202011-divvy-tripdata.csv")
December_2020 <- read.csv("../input/cyclistic-data/202012-divvy-tripdata.csv")
January_2021 <- read.csv("../input/cyclistic-data/202101-divvy-tripdata.csv")
February_2021 <- read.csv("../input/cyclistic-data/202102-divvy-tripdata.csv")
March_2021 <- read.csv("../input/cyclistic-data/202103-divvy-tripdata.csv")

In [65]:
#to know the structure data frames created 
str(April_2020)
str(January_2021)

'data.frame':	84776 obs. of  13 variables:
 $ ride_id           : chr  "A847FADBBC638E45" "5405B80E996FF60D" "5DD24A79A4E006F4" "2A59BBDF5CDBA725" ...
 $ rideable_type     : chr  "docked_bike" "docked_bike" "docked_bike" "docked_bike" ...
 $ started_at        : chr  "2020-04-26 17:45:14" "2020-04-17 17:08:54" "2020-04-01 17:54:13" "2020-04-07 12:50:19" ...
 $ ended_at          : chr  "2020-04-26 18:12:03" "2020-04-17 17:17:03" "2020-04-01 18:08:36" "2020-04-07 13:02:31" ...
 $ start_station_name: chr  "Eckhart Park" "Drake Ave & Fullerton Ave" "McClurg Ct & Erie St" "California Ave & Division St" ...
 $ start_station_id  : int  86 503 142 216 125 173 35 434 627 377 ...
 $ end_station_name  : chr  "Lincoln Ave & Diversey Pkwy" "Kosciuszko Park" "Indiana Ave & Roosevelt Rd" "Wood St & Augusta Blvd" ...
 $ end_station_id    : int  152 499 255 657 323 35 635 382 359 508 ...
 $ start_lat         : num  41.9 41.9 41.9 41.9 41.9 ...
 $ start_lng         : num  -87.7 -87.7 -87.6 -87.7 -87.6 ..

We saw how the data for April 2020 and January 2021 is organized.
As the dataset is too long, we will compare column data types across all data frames
using **compare_df_cols**

In [66]:
#comparing column data types among all data frames and get output of only mismached columns
compare_df_cols(April_2020,May_2020,June_2020,July_2020,August_2020,September_2020,October_2020,November_2020,December_2020,
                January_2021,February_2021,March_2021, return = "mismatch")

column_name      April_2020 May_2020 June_2020 July_2020 August_2020
1 end_station_id   integer    integer  integer   integer   integer    
2 start_station_id integer    integer  integer   integer   integer    
  September_2020 October_2020 November_2020 December_2020 January_2021
1 integer        integer      integer       character     character   
2 integer        integer      integer       character     character   
  February_2021 March_2021
1 character     character 
2 character     character

On comparing column data types of all data frames found some columns do not have the same data type.

Therefore, before merging all data frames into a single data frame we have make number of columns and column data types same.
Now, have to convert start_station_id and end_station_id columns of December_2020, January_2021, February_2021 and March_2021 as integer data type.

In [67]:

# warnings : as we have replaced missing/blank data with NA's, therefore to supress warning
options(warn=-1)

#converting character data type of end station column as integer data type

December_2020$end_station_id <- as.integer(December_2020$end_station_id)
January_2021$end_station_id <- as.integer(January_2021$end_station_id)
February_2021$end_station_id <- as.integer(February_2021$end_station_id)
March_2021$end_station_id <- as.integer(March_2021$end_station_id)

#converting character data type of start station column as integer data type

December_2020$start_station_id <- as.integer(December_2020$start_station_id)
January_2021$start_station_id <- as.integer(January_2021$start_station_id)
February_2021$start_station_id <- as.integer(February_2021$start_station_id)
March_2021$start_station_id <- as.integer(March_2021$start_station_id)

# let's get the glimpse of changes made
glimpse(December_2020)
glimpse(January_2021)
glimpse(February_2021)
glimpse(March_2021)

Rows: 131,573
Columns: 13
$ ride_id            <chr> "70B6A9A437D4C30D", "158A465D4E74C54A", "5262016E0F…
$ rideable_type      <chr> "classic_bike", "electric_bike", "electric_bike", "…
$ started_at         <chr> "2020-12-27 12:44:29", "2020-12-18 17:37:15", "2020…
$ ended_at           <chr> "2020-12-27 12:55:06", "2020-12-18 17:44:19", "2020…
$ start_station_name <chr> "Aberdeen St & Jackson Blvd", "", "", "", "", "", "…
$ start_station_id   <int> 13157, NA, NA, NA, NA, NA, NA, NA, NA, NA, NA, NA, …
$ end_station_name   <chr> "Desplaines St & Kinzie St", "", "", "", "", "", ""…
$ end_station_id     <int> NA, NA, NA, NA, NA, NA, NA, NA, NA, NA, NA, NA, NA,…
$ start_lat          <dbl> 41.87773, 41.93000, 41.91000, 41.92000, 41.80000, 4…
$ start_lng          <dbl> -87.65479, -87.70000, -87.69000, -87.70000, -87.590…
$ end_lat            <dbl> 41.88872, 41.91000, 41.93000, 41.91000, 41.80000, 4…
$ end_lng            <dbl> -87.64445, -87.70000, -87.70000, -87.70000, -87.590…
$ member_casua

Let's recheck column data types across all data frames

In [68]:
# checking again
#comparing column data types among all data frames and get output of only mismached columns
compare_df_cols(April_2020,May_2020,June_2020,July_2020,August_2020,September_2020,October_2020,November_2020,December_2020,
                January_2021,February_2021,March_2021, return = "mismatch")

column_name April_2020 May_2020 June_2020 July_2020 August_2020
     September_2020 October_2020 November_2020 December_2020 January_2021
     February_2021 March_2021

Now, we can merge all data frames into one data frame.

In [69]:
###--- now, no. of columns are same and columns across all data frames are same----

## merging all month's dataframe into a single dataframe

Trips20_21 <- rbind(April_2020,May_2020,June_2020,July_2020,August_2020,September_2020,October_2020,November_2020,
                    December_2020,January_2021,February_2021,March_2021)

Now, let's get the structure of single dataframe created

In [70]:
str(Trips20_21)

'data.frame':	3489748 obs. of  13 variables:
 $ ride_id           : chr  "A847FADBBC638E45" "5405B80E996FF60D" "5DD24A79A4E006F4" "2A59BBDF5CDBA725" ...
 $ rideable_type     : chr  "docked_bike" "docked_bike" "docked_bike" "docked_bike" ...
 $ started_at        : chr  "2020-04-26 17:45:14" "2020-04-17 17:08:54" "2020-04-01 17:54:13" "2020-04-07 12:50:19" ...
 $ ended_at          : chr  "2020-04-26 18:12:03" "2020-04-17 17:17:03" "2020-04-01 18:08:36" "2020-04-07 13:02:31" ...
 $ start_station_name: chr  "Eckhart Park" "Drake Ave & Fullerton Ave" "McClurg Ct & Erie St" "California Ave & Division St" ...
 $ start_station_id  : int  86 503 142 216 125 173 35 434 627 377 ...
 $ end_station_name  : chr  "Lincoln Ave & Diversey Pkwy" "Kosciuszko Park" "Indiana Ave & Roosevelt Rd" "Wood St & Augusta Blvd" ...
 $ end_station_id    : int  152 499 255 657 323 35 635 382 359 508 ...
 $ start_lat         : num  41.9 41.9 41.9 41.9 41.9 ...
 $ start_lng         : num  -87.7 -87.7 -87.6 -87.7 -87.6 

****DATA CLEANING****


Next, decided to removed missing/blank values to make the merging all CSV files into one dataframe easier.

In [71]:
Trips20_21 <- janitor::remove_empty(Trips20_21,which=c("cols"))
Trips20_21 <- janitor::remove_empty(Trips20_21,which=c("rows"))

Let's remove duplicated ride IDs

In [72]:
Trips20_21_1 <- Trips20_21[!duplicated(Trips20_21$ride_id), ]
print(paste("Removed", nrow(Trips20_21_1) - nrow(Trips20_21), "duplicated rows"))

[1] "Removed -209 duplicated rows"


We see start time and end time are character data type , therefore changing data type to date/time format.

In [73]:
### convert date/timestamp to date/time
Trips20_21$started_at <- strptime(Trips20_21$started_at,"%Y-%m-%d %H:%M:%S")
Trips20_21$ended_at <- strptime(Trips20_21$ended_at,"%Y-%m-%d %H:%M:%S")
str(Trips20_21)

'data.frame':	3489748 obs. of  13 variables:
 $ ride_id           : chr  "A847FADBBC638E45" "5405B80E996FF60D" "5DD24A79A4E006F4" "2A59BBDF5CDBA725" ...
 $ rideable_type     : chr  "docked_bike" "docked_bike" "docked_bike" "docked_bike" ...
 $ started_at        : POSIXlt, format: "2020-04-26 17:45:14" "2020-04-17 17:08:54" ...
 $ ended_at          : POSIXlt, format: "2020-04-26 18:12:03" "2020-04-17 17:17:03" ...
 $ start_station_name: chr  "Eckhart Park" "Drake Ave & Fullerton Ave" "McClurg Ct & Erie St" "California Ave & Division St" ...
 $ start_station_id  : int  86 503 142 216 125 173 35 434 627 377 ...
 $ end_station_name  : chr  "Lincoln Ave & Diversey Pkwy" "Kosciuszko Park" "Indiana Ave & Roosevelt Rd" "Wood St & Augusta Blvd" ...
 $ end_station_id    : int  152 499 255 657 323 35 635 382 359 508 ...
 $ start_lat         : num  41.9 41.9 41.9 41.9 41.9 ...
 $ start_lng         : num  -87.7 -87.7 -87.6 -87.7 -87.6 ...
 $ end_lat           : num  41.9 41.9 41.9 41.9 42 ...
 $ en

Separate column fo start and end date of trip

In [74]:
#create separte start and end date
Trips20_21$Date <- as.Date(Trips20_21$started_at) 
Trips20_21$start_date <- as.Date(Trips20_21$started_at) 
Trips20_21$end_date <-as.Date(Trips20_21$ended_at)
head(Trips20_21,5)      #starting 5 rows

ride_id          rideable_type started_at          ended_at           
1 A847FADBBC638E45 docked_bike   2020-04-26 17:45:14 2020-04-26 18:12:03
2 5405B80E996FF60D docked_bike   2020-04-17 17:08:54 2020-04-17 17:17:03
3 5DD24A79A4E006F4 docked_bike   2020-04-01 17:54:13 2020-04-01 18:08:36
4 2A59BBDF5CDBA725 docked_bike   2020-04-07 12:50:19 2020-04-07 13:02:31
5 27AD306C119C6158 docked_bike   2020-04-18 10:22:59 2020-04-18 11:15:54
  start_station_name           start_station_id end_station_name           
1 Eckhart Park                  86              Lincoln Ave & Diversey Pkwy
2 Drake Ave & Fullerton Ave    503              Kosciuszko Park            
3 McClurg Ct & Erie St         142              Indiana Ave & Roosevelt Rd 
4 California Ave & Division St 216              Wood St & Augusta Blvd     
5 Rush St & Hubbard St         125              Sheridan Rd & Lawrence Ave 
  end_station_id start_lat start_lng end_lat end_lng  member_casual Date      
1 152            41.8964   -87.6610  41.9322 -87.6586 member        2020-04-26
2 499            41.9244   -87.7154  41.9306 -87.7238 member        2020-04-17
3 255            41.8945   -87.6179  41.8679 -87.6230 member        2020-04-01
4 657            41.9030   -87.6975  41.8992 -87.6722 member        2020-04-07
5 323            41.8902   -87.6262  41.9695 -87.6547 casual        2020-04-18
  start_date end_date  
1 2020-04-26 2020-04-26
2 2020-04-17 2020-04-17
3 2020-04-01 2020-04-01
4 2020-04-07 2020-04-07
5 2020-04-18 2020-04-18

Additionally, for analysis process we need Month,day and weekday as separate columns

In [75]:
# create columns for month,day and year for all rides
Trips20_21$Month <- format(as.Date(Trips20_21$started_at),"%b")
Trips20_21$day <-format(as.Date(Trips20_21$started_at),"%d")
Trips20_21$Year <-format(as.Date(Trips20_21$started_at),"%y")
Trips20_21$weekday <-weekdays(Trips20_21$Date)
head(Trips20_21,5)

ride_id          rideable_type started_at          ended_at           
1 A847FADBBC638E45 docked_bike   2020-04-26 17:45:14 2020-04-26 18:12:03
2 5405B80E996FF60D docked_bike   2020-04-17 17:08:54 2020-04-17 17:17:03
3 5DD24A79A4E006F4 docked_bike   2020-04-01 17:54:13 2020-04-01 18:08:36
4 2A59BBDF5CDBA725 docked_bike   2020-04-07 12:50:19 2020-04-07 13:02:31
5 27AD306C119C6158 docked_bike   2020-04-18 10:22:59 2020-04-18 11:15:54
  start_station_name           start_station_id end_station_name           
1 Eckhart Park                  86              Lincoln Ave & Diversey Pkwy
2 Drake Ave & Fullerton Ave    503              Kosciuszko Park            
3 McClurg Ct & Erie St         142              Indiana Ave & Roosevelt Rd 
4 California Ave & Division St 216              Wood St & Augusta Blvd     
5 Rush St & Hubbard St         125              Sheridan Rd & Lawrence Ave 
  end_station_id start_lat start_lng end_lat end_lng  member_casual Date      
1 152            41.8964   -87.6610  41.9322 -87.6586 member        2020-04-26
2 499            41.9244   -87.7154  41.9306 -87.7238 member        2020-04-17
3 255            41.8945   -87.6179  41.8679 -87.6230 member        2020-04-01
4 657            41.9030   -87.6975  41.8992 -87.6722 member        2020-04-07
5 323            41.8902   -87.6262  41.9695 -87.6547 casual        2020-04-18
  start_date end_date   Month day Year weekday  
1 2020-04-26 2020-04-26 Apr   26  20   Sunday   
2 2020-04-17 2020-04-17 Apr   17  20   Friday   
3 2020-04-01 2020-04-01 Apr   01  20   Wednesday
4 2020-04-07 2020-04-07 Apr   07  20   Tuesday  
5 2020-04-18 2020-04-18 Apr   18  20   Saturday

In [76]:
#order days of the week 
Trips20_21$weekday <- ordered(Trips20_21$weekday, levels=c("Sunday","Monday","Tuesday","Wednesday","Thursday","Friday","Saturday"))
Trips20_21$Month <- ordered(Trips20_21$Month, levels=c('Jan','Feb','Mar','Apr','May','Jun','Jul','Aug','Sep','Oct','Nov','Dec'))


****MANIPULATING DATA****

To find out at which hour of the day rides started.

In [77]:
Trips20_21$start_hour <-  strftime(Trips20_21$started_at, "%H")
#head(Trips)
unique(Trips20_21$start_hour)

[1] "17" "12" "10" "14" "15" "18" "13" "02" "16" "08" "20" "19" "11" "23" "06"
[16] "09" "07" "22" "21" "01" "05" "00" "04" "03"

Now, let's check the duration of trips. Finding out the time difference between end time and start time would give us trip duration.

In [78]:

#new column - duration and checking for trip duartion in mins less than 0
## accuracy check
Trips20_21 <- mutate(Trips20_21,duration_min=as.numeric(Trips20_21$ended_at-Trips20_21$started_at)/60)
head(Trips20_21)
filter(Trips20_21, duration_min < 0)

ride_id          rideable_type started_at          ended_at           
1 A847FADBBC638E45 docked_bike   2020-04-26 17:45:14 2020-04-26 18:12:03
2 5405B80E996FF60D docked_bike   2020-04-17 17:08:54 2020-04-17 17:17:03
3 5DD24A79A4E006F4 docked_bike   2020-04-01 17:54:13 2020-04-01 18:08:36
4 2A59BBDF5CDBA725 docked_bike   2020-04-07 12:50:19 2020-04-07 13:02:31
5 27AD306C119C6158 docked_bike   2020-04-18 10:22:59 2020-04-18 11:15:54
6 356216E875132F61 docked_bike   2020-04-30 17:55:47 2020-04-30 18:01:11
  start_station_name                  start_station_id
1 Eckhart Park                         86             
2 Drake Ave & Fullerton Ave           503             
3 McClurg Ct & Erie St                142             
4 California Ave & Division St        216             
5 Rush St & Hubbard St                125             
6 Mies van der Rohe Way & Chicago Ave 173             
  end_station_name            end_station_id start_lat start_lng ⋯
1 Lincoln Ave & Diversey Pkwy 152            41.8964   -87.6610  ⋯
2 Kosciuszko Park             499            41.9244   -87.7154  ⋯
3 Indiana Ave & Roosevelt Rd  255            41.8945   -87.6179  ⋯
4 Wood St & Augusta Blvd      657            41.9030   -87.6975  ⋯
5 Sheridan Rd & Lawrence Ave  323            41.8902   -87.6262  ⋯
6 Streeter Dr & Grand Ave      35            41.8969   -87.6217  ⋯
  member_casual Date       start_date end_date   Month day Year weekday  
1 member        2020-04-26 2020-04-26 2020-04-26 Apr   26  20   Sunday   
2 member        2020-04-17 2020-04-17 2020-04-17 Apr   17  20   Friday   
3 member        2020-04-01 2020-04-01 2020-04-01 Apr   01  20   Wednesday
4 member        2020-04-07 2020-04-07 2020-04-07 Apr   07  20   Tuesday  
5 casual        2020-04-18 2020-04-18 2020-04-18 Apr   18  20   Saturday 
6 member        2020-04-30 2020-04-30 2020-04-30 Apr   30  20   Thursday 
  start_hour duration_min
1 17         26.81667    
2 17          8.15000    
3 17         14.38333    
4 12         12.20000    
5 10         52.91667    
6 17          5.40000

ride_id          rideable_type started_at          ended_at           
1     7C1E92200AEFF70E docked_bike   2020-04-27 17:20:30 2020-04-27 17:20:18
2     671BB1F73F4CD303 docked_bike   2020-04-20 16:51:18 2020-04-20 16:51:05
3     502B972C6B1FCAE6 docked_bike   2020-04-12 19:09:54 2020-04-12 19:09:40
4     68E70FDF06F0A439 docked_bike   2020-04-29 17:37:11 2020-04-29 17:37:07
5     6EB323BCC83A9D1D docked_bike   2020-04-05 15:46:12 2020-04-05 15:46:11
6     90105A0FA1F2B0F3 docked_bike   2020-04-13 14:06:59 2020-04-13 14:06:45
7     BDFF2212459A9858 docked_bike   2020-04-27 18:51:42 2020-04-27 18:51:14
8     BFF9D20C42D3B693 docked_bike   2020-04-19 14:10:16 2020-04-19 14:10:03
9     15FE83B5CC494A1C docked_bike   2020-04-19 18:52:35 2020-04-19 18:52:32
10    F6FDF112F975A216 docked_bike   2020-04-28 15:35:24 2020-04-28 15:35:11
11    00ED4786F962B827 docked_bike   2020-04-28 06:55:20 2020-04-28 06:54:33
12    3C19503CC3A81CCE docked_bike   2020-04-29 16:54:01 2020-04-29 16:51:05
13    83A8FE824A191902 docked_bike   2020-04-14 07:52:00 2020-04-14 07:51:48
14    B99C5A0F67262E04 docked_bike   2020-04-28 15:45:05 2020-04-28 15:45:00
15    ADC76DBE67BAAD66 docked_bike   2020-04-29 17:41:39 2020-04-29 17:41:33
16    DF401D495C8822E1 docked_bike   2020-04-24 16:19:11 2020-04-24 16:18:53
17    12608C026A6E01C4 docked_bike   2020-04-29 17:50:14 2020-04-29 17:50:02
18    41E80A7BDB6409C3 docked_bike   2020-04-24 17:59:12 2020-04-24 17:59:00
19    6FC11E831B21B28D docked_bike   2020-04-27 18:49:58 2020-04-27 18:49:25
20    D6092C5E242F6D47 docked_bike   2020-04-15 19:12:09 2020-04-15 19:12:04
21    72E721ADC38364D2 docked_bike   2020-04-14 07:53:26 2020-04-14 07:53:22
22    6F90CC047E2C55E2 docked_bike   2020-04-13 18:12:58 2020-04-13 18:12:44
23    05B000481136CAF5 docked_bike   2020-04-24 17:49:36 2020-04-24 17:49:19
24    11195C3052EE09B2 docked_bike   2020-04-11 16:39:48 2020-04-11 16:39:33
25    E28382CB814CAD8C docked_bike   2020-04-28 06:53:39 2020-04-28 06:53:32
26    C6F50A326A5F883E docked_bike   2020-04-19 18:51:46 2020-04-19 18:51:10
27    4C4F851B25D81BBC docked_bike   2020-04-29 13:34:01 2020-04-29 13:33:56
28    B32C8EF3CE9CC515 docked_bike   2020-04-01 14:22:11 2020-04-01 14:22:00
29    A9C677AB30627686 docked_bike   2020-04-15 19:15:44 2020-04-15 19:15:15
30    ED9B2819C03EAA96 docked_bike   2020-04-15 19:09:04 2020-04-15 19:08:52
⋮     ⋮                ⋮             ⋮                   ⋮                  
10523 84C6FB14707FAE6F docked_bike   2020-12-03 10:06:09 2020-12-03 10:05:45
10524 8FE484F4002A8D95 docked_bike   2020-12-15 11:55:07 2020-11-25 15:33:17
10525 CE697DE4AA389B16 docked_bike   2020-12-04 08:26:47 2020-12-04 08:26:18
10526 194FB266E3663610 docked_bike   2020-12-15 11:44:26 2020-11-25 13:35:56
10527 7C726A23F6D61BC3 electric_bike 2020-12-15 12:04:42 2020-11-25 18:44:42
10528 7247F76EE11C8B97 docked_bike   2020-12-15 11:57:00 2020-11-25 16:30:59
10529 9DBE8582AD0C08B8 electric_bike 2020-12-15 11:50:44 2020-11-25 15:40:45
10530 8321E86D4393271A electric_bike 2020-12-15 11:54:44 2020-11-25 17:00:16
10531 5078D72FF8A2357D docked_bike   2020-12-02 14:33:21 2020-12-02 14:32:47
10532 F1F427406BFB3C86 electric_bike 2020-12-15 12:01:09 2020-11-25 16:58:50
10533 B816A83C2E37ED0A electric_bike 2020-12-15 12:12:15 2020-11-25 22:50:26
10534 863CA750B639B627 electric_bike 2020-12-15 11:51:35 2020-11-25 15:16:59
10535 87D4C892CA85DB8F docked_bike   2020-12-15 11:54:25 2020-11-25 16:12:01
10536 84EEACD1FE641F30 docked_bike   2020-12-15 11:58:10 2020-11-25 17:07:25
10537 47E0C01E8F7BD830 docked_bike   2020-12-15 12:00:02 2020-11-25 18:06:03
10538 332340BB61D42B3C docked_bike   2020-12-15 12:21:37 2020-11-25 14:49:41
10539 6C1279812BD9D0C2 docked_bike   2020-12-15 12:08:00 2020-11-26 08:25:29
10540 5EE680AA8D46F8B8 docked_bike   2020-12-15 12:14:03 2020-11-25 09:23:46
10541 485D9BFF665E3987 docked_bike   2020-12-02 19:11:59 2020-12-02 19:11:18
10542 5CE6FB6301017AB2 electric_bike 2020-12-15 12:01:18 2020-11-25 18:09:23
10

Calculated field duration_min has values which are less than zero, which could be possibly errors. Therefore, have to remove rows where duration is less than zero.

In [79]:

# filtering to include rows with duration greater than 0
Trips20_21_1 <- filter(Trips20_21, Trips20_21$duration_min >0)
head(Trips20_21_1)

ride_id          rideable_type started_at          ended_at           
1 A847FADBBC638E45 docked_bike   2020-04-26 17:45:14 2020-04-26 18:12:03
2 5405B80E996FF60D docked_bike   2020-04-17 17:08:54 2020-04-17 17:17:03
3 5DD24A79A4E006F4 docked_bike   2020-04-01 17:54:13 2020-04-01 18:08:36
4 2A59BBDF5CDBA725 docked_bike   2020-04-07 12:50:19 2020-04-07 13:02:31
5 27AD306C119C6158 docked_bike   2020-04-18 10:22:59 2020-04-18 11:15:54
6 356216E875132F61 docked_bike   2020-04-30 17:55:47 2020-04-30 18:01:11
  start_station_name                  start_station_id
1 Eckhart Park                         86             
2 Drake Ave & Fullerton Ave           503             
3 McClurg Ct & Erie St                142             
4 California Ave & Division St        216             
5 Rush St & Hubbard St                125             
6 Mies van der Rohe Way & Chicago Ave 173             
  end_station_name            end_station_id start_lat start_lng ⋯
1 Lincoln Ave & Diversey Pkwy 152            41.8964   -87.6610  ⋯
2 Kosciuszko Park             499            41.9244   -87.7154  ⋯
3 Indiana Ave & Roosevelt Rd  255            41.8945   -87.6179  ⋯
4 Wood St & Augusta Blvd      657            41.9030   -87.6975  ⋯
5 Sheridan Rd & Lawrence Ave  323            41.8902   -87.6262  ⋯
6 Streeter Dr & Grand Ave      35            41.8969   -87.6217  ⋯
  member_casual Date       start_date end_date   Month day Year weekday  
1 member        2020-04-26 2020-04-26 2020-04-26 Apr   26  20   Sunday   
2 member        2020-04-17 2020-04-17 2020-04-17 Apr   17  20   Friday   
3 member        2020-04-01 2020-04-01 2020-04-01 Apr   01  20   Wednesday
4 member        2020-04-07 2020-04-07 2020-04-07 Apr   07  20   Tuesday  
5 casual        2020-04-18 2020-04-18 2020-04-18 Apr   18  20   Saturday 
6 member        2020-04-30 2020-04-30 2020-04-30 Apr   30  20   Thursday 
  start_hour duration_min
1 17         26.81667    
2 17          8.15000    
3 17         14.38333    
4 12         12.20000    
5 10         52.91667    
6 17          5.40000

We might have missing or blank values , let's remove those such data before getting into analysis.

In [80]:
nrow(Trips20_21_1)
Trips<-drop_na(Trips20_21_1)
nrow(Trips)

[1] 3478810

[1] 2934265

In [81]:
head(Trips,5)

ride_id          rideable_type started_at          ended_at           
1 A847FADBBC638E45 docked_bike   2020-04-26 17:45:14 2020-04-26 18:12:03
2 5405B80E996FF60D docked_bike   2020-04-17 17:08:54 2020-04-17 17:17:03
3 5DD24A79A4E006F4 docked_bike   2020-04-01 17:54:13 2020-04-01 18:08:36
4 2A59BBDF5CDBA725 docked_bike   2020-04-07 12:50:19 2020-04-07 13:02:31
5 27AD306C119C6158 docked_bike   2020-04-18 10:22:59 2020-04-18 11:15:54
  start_station_name           start_station_id end_station_name           
1 Eckhart Park                  86              Lincoln Ave & Diversey Pkwy
2 Drake Ave & Fullerton Ave    503              Kosciuszko Park            
3 McClurg Ct & Erie St         142              Indiana Ave & Roosevelt Rd 
4 California Ave & Division St 216              Wood St & Augusta Blvd     
5 Rush St & Hubbard St         125              Sheridan Rd & Lawrence Ave 
  end_station_id start_lat start_lng ⋯ member_casual Date       start_date
1 152            41.8964   -87.6610  ⋯ member        2020-04-26 2020-04-26
2 499            41.9244   -87.7154  ⋯ member        2020-04-17 2020-04-17
3 255            41.8945   -87.6179  ⋯ member        2020-04-01 2020-04-01
4 657            41.9030   -87.6975  ⋯ member        2020-04-07 2020-04-07
5 323            41.8902   -87.6262  ⋯ casual        2020-04-18 2020-04-18
  end_date   Month day Year weekday   start_hour duration_min
1 2020-04-26 Apr   26  20   Sunday    17         26.81667    
2 2020-04-17 Apr   17  20   Friday    17          8.15000    
3 2020-04-01 Apr   01  20   Wednesday 17         14.38333    
4 2020-04-07 Apr   07  20   Tuesday   12         12.20000    
5 2020-04-18 Apr   18  20   Saturday  10         52.91667

Arranging data in ascending order

In [82]:
Trips<-arrange(Trips,started_at)
head(Trips,5)

ride_id          rideable_type started_at          ended_at           
1 5DB63F4E4EB6A9CF docked_bike   2020-04-01 00:00:30 2020-04-01 00:23:03
2 1FD159E93F7BAFA1 docked_bike   2020-04-01 00:02:35 2020-04-01 00:10:45
3 6D93A270684EC452 docked_bike   2020-04-01 00:02:41 2020-04-01 00:24:20
4 091D47E4F0FC5022 docked_bike   2020-04-01 00:06:44 2020-04-01 00:14:01
5 07F785C9DDA3404C docked_bike   2020-04-01 00:11:18 2020-04-01 00:11:51
  start_station_name                  start_station_id
1 Damen Ave & Wellington Ave          162             
2 Wabash Ave & 16th St                 72             
3 Damen Ave & Wellington Ave          162             
4 Mies van der Rohe Way & Chicago Ave 173             
5 Wabash Ave & 9th St                 321             
  end_station_name              end_station_id start_lat start_lng ⋯
1 Pine Grove Ave & Waveland Ave 232            41.9359   -87.6784  ⋯
2 Wabash Ave & 9th St           321            41.8604   -87.6258  ⋯
3 Spaulding Ave & Armitage Ave  506            41.9359   -87.6784  ⋯
4 Clark St & Schiller St        301            41.8969   -87.6217  ⋯
5 Wabash Ave & 9th St           321            41.8708   -87.6257  ⋯
  member_casual Date       start_date end_date   Month day Year weekday  
1 casual        2020-04-01 2020-04-01 2020-04-01 Apr   01  20   Wednesday
2 member        2020-04-01 2020-04-01 2020-04-01 Apr   01  20   Wednesday
3 casual        2020-04-01 2020-04-01 2020-04-01 Apr   01  20   Wednesday
4 member        2020-04-01 2020-04-01 2020-04-01 Apr   01  20   Wednesday
5 member        2020-04-01 2020-04-01 2020-04-01 Apr   01  20   Wednesday
  start_hour duration_min
1 00         22.550000   
2 00          8.166667   
3 00         21.650000   
4 00          7.283333   
5 00          0.550000

****Saving clean data****

Saving the clean data as CSV file.

In [ ]:
Trips %>%
  write.csv("Trips.csv")
read.csv("Trips.csv")

**1.** **Guiding questions** 

* ***What tools are you choosing and why?***

I have chosen R for this project, as the dataset is quite large. MS Excel or Google spreadsheet was time-consuming for cleaning or sorting data. Therefore in Kaggle, I have opted to use R.

* ***Have you ensured your data’s integrity?***

Yes, ensured the data is consistent throughout the columns.

* ***What steps have you taken to ensure that your data is clean?***

Have removed missing/blank values and also checked for duplicate values, and also formatted columns to correct data type.

* ***How can you verify that your data is clean and ready to analyze?***

The processes carried out in Process step of data analysis in the R notebook of Kaggle are the only way to verify and ensure it is ready for analyze.

* ***Have you documented your cleaning process so you can review and share those results?***

Yes, all the cleaning process has been documented to review when required and ready to share results.

**2.** **Key tasks**

[x] Check the data for errors.

[x] Choose your tools.

[x] Transform the data so you can work with it effectively.

[x] Document the cleaning process.

**3.** **Deliverable**

[x] Documentation of any cleaning or manipulation of data.

# **Analyze** 
#### **[Click here](https://public.tableau.com/app/profile/roopa.d.moorthy/viz/GoogleDataAnalyticsCapstoneProjectCyclistic_16620301572380/GoogleDataAnalyticsCapstoneProject)** for data visualization on Tableau.

Checking types of Cyclistic riders and bike types included to analyze data.

In [ ]:
unique(Trips$member_casual)
unique(Trips$rideable_type)

let's see the summary of the dataset:

In [ ]:
summary(Trips)

In [ ]:
#all rides duraion summary
summary(Trips$duration_min)

**1.** ***Data Distribution :rider type***

* ****How much percentage is Annual members and casual riders?****

In [ ]:
Trips %>%
group_by(member_casual) %>%
summarise(Total_rides= length(ride_id),
         percentage= length(ride_id)/nrow(Trips)*100)

Annual members compose ~57% which is  ~15% greater than the current casual riders.

In [ ]:
Trips %>%
group_by(member_casual) %>%
summarise(avg_duration=mean(duration_min),median_duration=median(duration_min),max_duration=max(duration_min),
          min_duration=min(duration_min))

**2. Monthly analysis : Rider type**

In [ ]:
Trips %>%
group_by(Month) %>%
summarise(total_no_trips = length(ride_id),
          'Monthly_Trip_%_/_Annual_trip' = length(ride_id) / nrow(Trips) * 100,
         member_percentage=(sum(member_casual=='member')/ length(ride_id)) *100,
         casual_percentage=(sum(member_casual=='casual')/ length(ride_id)) * 100,
         percentage_difference= member_percentage-casual_percentage)

* The month with the maximum number of trips is August 2020, approx. 20% of total annual trips are from August 2020.
* Riders with annual membership (member in the output table) took more rides compared to casual riders.

As per Chicago climate from April to May is spring, June to August is Summer, September to November is Fall and December to February is Winter.

In [ ]:

Trips$Season <-ifelse((Trips$Month=='Mar')|(Trips$Month=='Apr')|(Trips$Month=='May'),'Spring',
                      ifelse((Trips$Month=='Jun')|(Trips$Month=='Jul')|(Trips$Month=='Aug'),'Summer',
                         ifelse((Trips$Month=='Sept')|(Trips$Month=='Oct')|(Trips$Month=='Nov'),'Fall','Winter' )) 
                     )
unique(Trips$Season)
head(Trips)

In [ ]:
Trips %>%
group_by(Season) %>%
summarise(total_no_trips = length(ride_id),
          'Season_Trip_%_/_Annual_trip' = length(ride_id) / nrow(Trips) * 100,
         member_percentage=(sum(member_casual=='member')/ length(ride_id)) *100,
         casual_percentage=(sum(member_casual=='casual')/ length(ride_id)) * 100,
         percentage_difference= member_percentage-casual_percentage)

* We can see 50% of total rides of year were during Summer.
* Summer months : June,July and August, and we also know in the month of August there were more rides compared to other months of the year.

**3.Weekly Analysis**

Data Distribution by days of the week

In [ ]:
Trips$start_hour <-  strftime(Trips$started_at, "%H")
#head(Trips)
unique(Trips$start_hour)

In [ ]:
Trips %>%
group_by(start_hour) %>%
summarise(total_no_trips = length(ride_id),
          'Hour_%' = length(ride_id) / nrow(Trips) * 100,
         member_percentage=(sum(member_casual=='member')/ length(ride_id)) *100,
         casual_percentage=(sum(member_casual=='casual')/ length(ride_id)) * 100,
         percentage_difference= member_percentage-casual_percentage)

* Most bikers prefered afternoon time.

* Annual members peak time : 5 am to 11 am and 5 pm to 6 pm.

* Casual members peak time: afternoon- 11 am to 4 pm,evening-6pm to 7 pm, late nights- 8 pm to 11 pm and early morning:12 am to 4 am.


For further analysis of data more quicker and to have interactive visualzation, I have used **[Tableau](https://public.tableau.com/app/profile/roopa.d.moorthy/viz/GoogleDataAnalyticsCapstoneProjectCyclistic_16620301572380/GoogleDataAnalyticsCapstoneProject)**.

**1. Guiding questions**

* **How should you organize your data to perform analysis on it?**

To perform analysis the data has been organized into a CSV file by concatenating all the files from the dataset.

* **Has your data been properly formatted?**

Yes, as all the columns have the correct data type. Hence, the data is properly formatted

* **What surprises did you discover in the data?**

Surprises from the data: Casual ride trends differ among Annual members on weekdays and hours of the day. Also, they spend more time on rides compared to Annual members.

* **What trends or relationships did you find in the data?**

More Annual members than casuals in the dataset.
Hour trends by week displayed the difference between casual and annual members on weekends.
Both casual and annual members took more rides on the docked bike.
Members to/from popular stations made trips in the morning while casuals took rides to/from the popular station.
Annual ride duration and number of rides for both ride types decreased in the winter season.

* **How will these insights help answer your business questions?**

The insights will help to understand the usage patterns of both riders.

**2.Key tasks**

[x] Aggregate your data so it’s useful and accessible.

[x] Organize and format your data.

[x] Perform calculations.

[x] Identify trends and relationships.

**3. Deliverable**

[x] A summary of your analysis

# Share

For the share phase,  the CSV file "Trips.csv" after cleaning is visualized in [Tableau for presentation](https://public.tableau.com/app/profile/roopa.d.moorthy/viz/GoogleDataAnalyticsCapstoneProjectCyclistic_16620301572380/GoogleDataAnalyticsCapstoneProject).
Here on Kaggle, the best way is its notebook for analysis and conclusions.

From the dataset:

After removing missing/blank/null values 

* The total number of rides : 2,934,265
* More Annual members than casual riders, 14.38 % more than casual riders.
* The month with the maximum number of rides is August (20.64 % ) - 605652 total rides.
* Riders type on the total number of rides each month: More annual member's rides than casual member's rides.
* Seasons influence the number of rides, the winter season recorded less number of rides.
* Weekends experienced a huge volume of rides.
* The volume of bikers increases in the afternoon.

**Why more annual members than casual riders?**

Annual members have more need for bikes than casual riders probably for commuting to work and also annual members took more rides than casual members during the winter season.
Whereas casual riders have more needs during summer and on weekends for a longer duration may be for recreational activities.


**How Annual members differ from casual riders:**

Annual members took more rides from Monday-Friday, and casual members took more rides on Saturday and Sunday.
Casual riders took more rides on weekends, ~10 % increase.
Annual members took more rides from 5 am -8 am, the huge spike can be seen around 7 am-8 am, noon and evening 5-7 pm. Casual riders between 11 AM - 4 PM And at 5- 7 PM.
During the weekend casuals had rides between 11 AM to 6 PM.
Casual has a preference for a docked and electric bike. Annual members have a preference for a classic bike.
The average ride duration of casual riders is more than annual members.
Annual member's ride duration is stable until midweek and increases on weekends.


With this, we can conclude that Annual members need bikes for a fixed purpose. They use bikes for :
* commute to work
* exercise

whereas, casual riders may be on ride for
* Visitng places/Tourists spots


Hence, we can say Annual members took bike rides in the morning 7-8 AM for work and also in the evening 5-6 PM may be because to get back to their place after work.Their ride time remained unchanged during weekdays as they have fixed route to work and ride time changed during weekends because they took bike for recreational activities.




**1. Guiding questions**

* **Were you able to answer the question of how annual members and casual riders use Cyclistic bikes differently?**

Yes, the data points helped to get trends /relationship to build profile of both riders.

* **What story does your data tell?**

The story is Annual members took rides for scheduled tasks like commuting to work on weekdays and on weekends for recreational activities and also they had less ride duration as they took set of routes.
But casual members, most rides were in the afternoon and evening time.They took rides for longer duration and especially more rides on weekends.

* **How do your findings relate to your original question?**

The findings is beneficial to build profile of the riders which relates to " Understanding both riders bike usage patterns" and with the information of when they use bikes more often helps with "creating marketing startegies".

* **Who is your audience? What is the best way to communicate with them?**

The audiences are cyclistic marketing analytics team, cyclistic executive team  and Lily Moreno. The best way to commuinicate is through data findings slide presentation.

* **Can data visualization help you share your findings?**

Yes, the data visualizations is carried out using interactive dashboards on tableau ,great tool for data visualizations.

* **Is your presentation accessible to your audience?**

Yes, the data plots are made using appropriate contrast, and corresponding labels.

**2. Key tasks**

[x] Determine the best way to share your findings.

[x] Create effective data visualizations.

[x] Present your findings.

[x] Ensure your work is accessible.

**3. Deliverable**

[x] Supporting visualizations and key findings.







# Act

The act phase is carried out by the marketing team of the copmpany which also includes deciding whether to approve recommended marketing programs based on findings.

The top 3 recommendations based on findings:

* 1. About **~ 60.87 % Casual riders took more rides during July- September** which are considered months of events and festivals in Chicago city. It is advisable to **build marketing campaigns focusing on the top 10 stations in the month July-September**. Using **social media to post about contests like - #daretoride** will attract casual riders and create an urge to avail benefits of added membership passes by winning contests.

Different pricing strategies like **weekly passes and monthly/seasonal passes** might improve conversion rates.

* 2. Also to create an experience **awarding discounts on cryptocurrencies or tokens as benefits, based on bike usage for those casual riders who avail membership**.

* 3. Mobile application for membership holders: **Building cyclistic mobile applications with fitness trackers and, also with live local traffic information and weather updates would add more value to membership passes** and increase conversion rates.

**1. Guiding questions**

* What is your final conclusion based on your analysis?
Members and casual riders have cyclistic bike usage patterns are different .

* How could your team and business apply your insights?.
The insights shows the trend by hour ,weekdays ,months/season of each rider type which could help to create effective marketing campaign to convert casual riders into annual members.

* What next steps would you or your stakeholders take based on your findings?
My stakeholders might further analyze or verify the findings to create effective marketing campaign within planned budget.

**2.Key tasks**

[x] Create your portfolio.

[x] Add your case study.

[x] Practice presenting your case study to a friend or family member.

**3.Deliverable**

[x] Your top three recommendations based on your analysis.




Finally, I would like to thank the[ Google Data Analytics Professional Certificate Coursera team](https://www.coursera.org/professional-certificates/google-data-analytics?utm_source=gg&utm_medium=sem&utm_campaign=15-GoogleDataAnalytics-LATAM&utm_content=15-GoogleDataAnalytics-LATAM&campaignid=12686019520&adgroupid=120140812253&device=c&keyword=google%20analytics%20course&matchtype=b&network=g&devicemodel=&adpostion=&creativeid=512414119178&hide_mobile_promo&gclid=EAIaIQobChMIwPGyn7f68AIVAgyRCh2nCwfVEAAYAyAAEgKjpvD_BwE) for creating such a wonderful journey of learning Data Analytics which even gave me an opportunity to apply my learnings to this project. Also, Thank you R community and Kaggle to help me use interfaces seamlessly.
